### PR TITLE
edit instructions & fix context.flags.app & fix config:set options

### PR DESCRIPTION
**Pull request by RasPhilCo**

### CODE REVIEW

File: src/commands/autocomplete/options.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export default class AutocompleteOptions extends AutocompleteBase {
   private get parsedFlagsWithEnvVars() {
     return Object.assign(
       {
-        app: process.env.HEROKU_APP,
+        app: process.env.HEROKU_APP || this.flags.app,
```
Comment (RasPhilCo) [inline src/commands/autocomplete/options.ts]: @crcastle this fixes the app issue when inside a project, no need for `--app=` anymore to complete

File: src/commands/autocomplete/index.test.ts [hunk 2]
```diff
@@ -10,76 +10,49 @@ cli.config.mock = true
...
 runtest('outputs install instructions for zsh', async () => {
   await Autocomplete.mock(['zsh'])
-  expect(cli.stdout.output).toMatch(`Setup Instructions for CLI-ENGINE CLI Autocomplete ---
-
-1) Add the autocomplete env var to your zsh profile
-
-$ printf "$(cli-engine autocomplete:script zsh)" >> ~/.zshrc
-
-2) Source your updated zsh profile
+  expect(cli.stdout.output).toMatch(`
```
Comment (RasPhilCo) [inline src/commands/autocomplete/index.test.ts]: @crcastle new zsh/bash output (cache build has been moved to the top but you won't see it here in the test)

File: src/commands/autocomplete/index.ts [hunk 3]
```diff
@@ -22,54 +22,38 @@ export default class Autocomplete extends AutocompleteBase {
...
       cli.error('Error: Missing required argument shell')
+    } else if (shell !== 'bash' && shell !== 'zsh') {
+      cli.error(`Currently ${shell} is not a supported shell for autocomplete`)
     }
 
-    if (!this.flags['skip-instructions']) {
-      if (shell !== 'bash' && shell !== 'zsh') {
-        cli.error(`Currently ${shell} is not a supported shell for autocomplete`)
-      }
+    cli.action.start(`${cli.color.bold('Building the autocomplete cache')}`)
```
Comment (RasPhilCo) [inline src/commands/autocomplete/index.ts]: `'Building the autocomplete cache...<spinner.'` @crcastle feel free to suggest different verbiage

File: src/autocomplete.ts [hunk 4]
```diff
@@ -23,12 +23,28 @@ const ConfigCompletion: flags.ICompletion = {
...
+const ConfigSetCompletion: flags.ICompletion = {
+  cacheDuration: 60 * 60 * 24 * 7,
+  cacheKey: async (ctx: any) => {
+    return ctx.flags && ctx.flags.app ? `${ctx.flags.app}_config_set_vars` : ''
+  },
+  options: async (ctx: any) => {
+    const heroku = new APIClient(ctx.config)
+    if (ctx.flags && ctx.flags.app) {
+      let { body: configs } = await heroku.get(`/apps/${ctx.flags.app}/config-vars`)
+      return Object.keys(configs).map(k => `${k}=`)
```
Comment (RasPhilCo) [inline src/autocomplete.ts]: @jdxcode now `config:set` completions will have `=` at the end

**Comment by crcastle** [comment]
Nice, looks like everything we talked about! Only thing I noticed is the slug on the draft dev center article (heroku-cli-command-completion) differs from what you have in the URL you added to the instructions.  I don't have an opinion on what it is -- just know it needs to match.

Other than that, looks good.  Although, you probably don't want me being the one approving this 😛 since I'm unfamiliar with the codebase.

**Pull request merged**